### PR TITLE
Fixes to Kenlm CMakeLists for succesfull compilation on Ubuntu, with static boost linking and with old systems

### PR DESCRIPTION
**Pull request by gwenniger**
1. Kenlm does not compile properly because zlib is not found (on ubuntu)
      => Fixed by adding zlib find and include library commands to the root CMakeKist file
2. On old g++ compilers the compilation fails because lib "real time", sometimes known
    as "-lrt" or simply "rt" was not included.
     => Added a fix to include this library.
3. On certain systems we need to link boost statically, making this work however
    required both a change in the CMakeLists root file, and also a change in
    kenlm/cmake/KenLMFunctions.cmake where the option -DBOOST_TEST_DYN_LINK
    should not be added when static binding to boost is used.
   
   ```
   modified:   ext/kenlm/CMakeLists.txt
   modified:   ext/kenlm/cmake/KenLMFunctions.cmake
   ```

### CODE REVIEW

**Comment by mjpost** [comment]
Hi Gideon,

Thanks for these! I think, however, that the more appropriate place for these patches is against [KenLM proper](https://github.com/kpu/kenlm/), since `$JOSHUA/ext/kenlm` is just a direct copy of that repo. This is also good to do because we intend to soon switch over to pulling it in as a submodule.

So, please, send the patches there, and then we will pull them in when they're fixed upstream.

**Comment by gwenniger** [comment]
Hi Matt,
Thanks. I will do as you suggested.

Gideon

**Comment by gwenniger** [comment]
Hi Matt,
I tried the latest KenLM version on our server.
After turning on the option

*    "option(FORCE_STATIC "Build static executables" ON)"*
and commenting out the line
* set(CMAKE_FIND_LIBRARY_SUFFIXES ".a")*
in the "FORCE_STATIC" block
it did work.
These remaining changes are not bugs but rather just requirements for our
server, with it's strange combination of outdated system libraries and
locally
installed boost etc.
The problems I encountered were already fixed in very recent commits to
kenlm proper it seems.
In summary, currently there is no need to file a kenlm bug report anymore,
just updating in Joshua to the
latest kenlm version as already in the planning I understand, is sufficient.
Cheers,

Gideon

**Comment by KellenSunderland** [comment]
Hey Gideon, just wanted to give a +1 that I was having the same issue.  I think your code fix would have addressed my issues, but merging from kenlm may also fix the problem.  Either way thanks for contributing.

**Comment by mjpost** [comment]
Gideon, did you submit this to KenLM? It'd be great if you'd do that and then post a link here so that we can track it.

**Comment by gwenniger** [comment]
Hi Matt,
My conclusion was that indeed this has now already been fixed in a recent
commit to the original kenlm version on github.
The version of kenlm in Joshua is behind on this latest version however,
and needs to be updated (or replaced by kenlm as
a directly tracked submodule, as you mentioned) to fix the problem.
I did not make a bug report because the bug was already fixed in kenlm
proper,  they have encountered the same problems and made a solution
very similar to the fix in my original pull-request, and therefore a a bug
report would not make sense anymore
as far as I can see.
Cheers.

Gideon

**Comment by mjpost** [comment]
Oh, great, thanks for the update. I'll pull the newer KenLM in soon.
